connection_testcases.json

```diff
@@ -2255,5 +2255,63 @@
     "error": {
       "type":"invalid_credentials_file"
     }
+  },
+  {
+    "opts": {
+      "dsn": "edgedb://user3:123123@[::1]:5555/abcdef"
+    },
+    "result": {
+      "address": ["::1", 5555],
+      "database": "abcdef",
+      "password": "123123",
+      "serverSettings": {},
+      "tlsCAData": null,
+      "tlsSecurity": "strict",
+      "user": "user3"
+    }
+  },
+  {
+    "opts": {
+      "dsn": "edgedb://user3@[fe80::1ff:fe23:4567:890a%25eth0]:3000/ab"
+    },
+    "result": {
+      "address": ["fe80::1ff:fe23:4567:890a%25eth0", 3000],
```

Comment:
This should be `fe80::1ff:fe23:4567:890a%eth0`, you need to `urllib.parse.unquote` it

Reply:
oooh good catch!

Reply:
I've updated the tests -- please review again